pypeit/specutils/pypeit_loaders.py

```diff
@@ -35,6 +37,69 @@
 from pypeit import onespec
 
 
+def enforce_monotonic_wavelengths(wave, flux, ivar, strict=True):
+    """
+    Force the spectrum to have a monotonically increasing wavelength vector.
+
+    Parameters
+    ----------
+    wave : `numpy.ndarray`_
+        Spectrum wavelengths
+    flux : `numpy.ndarray`_
+        Spectrum flux
+    ivar : `numpy.ndarray`_
+        Spectrum inverse variance. Can be None.
+    strict : bool, optional
+        Check that the wavelength vector is monotonically increasing.  If not,
+        raise an error (as would be done by the `specutils.SpectrumList`_ class).
+        If False, wavelengths that are *not* monotonically increasing are masked
+        in the construction of the returned `specutils.SpectrumList`_ object.
+
+    Returns
+    -------
+    wave : `numpy.ndarray`_
+        Edited wavelength vector.  This may be an unchanged reference to the
+        original vector.
+    flux : `numpy.ndarray`_
+        Edited flux vector.  This may be an unchanged reference to the original
+        vector.
+    ivar : `numpy.ndarray`_
+        Edited inverse variance vector.  This may be an unchanged reference to
+        the original vector.
+    """
+    indx = np.diff(wave) > 0
+    if np.all(indx):
+        # Wavelengths are monotonic, so we're done
+        return wave, flux, ivar
+
+    if strict:
+        # Wavelengths are not monotonic, but the user expects them to be, so
+        # fault.
+        msgs.error('Wavelengths are not monotonically increasing!  Circumvent this fault by '
+                   'setting strict=False, but BEWARE that this is likely the result of an '
+                   'error in the data reduction!')
+
+    # Wavelengths are not monotonic, but the user wants to keep going.
+    msgs.warn('Wavelengths are not monotonically increasing!  Strict was set to False, so '
+              'measurements after a negative step in wavelength are removed from the constructed '
+              'spectrum.  BEWARE that this is likely the result of an error in the data '
+              'reduction!')
+
+    # NOTE: This is the brute force approach.  If this becomes something that we
+    # want to be more acceptable, we should consider instead fitting a low-order
+    # polynomial to the pixel vs. wavelength function and rejecting strong
+    # outliers.
+    _wave = wave.copy()
```

Comment:
I think all of the `wave` below this line should be changed to `_wave`

EDIT: Actually it is probably fine, since you pass in and set `wave` in `pypeit_onespec_loader()`... but I still flag this just in case

Reply:
I think I agree with @rcooke-ast 's first comment, else `_wave` doesn't do anything here.

Reply:
Good catch, and apologies.  This was a leftover line I used during testing.  Removing the line fixes the issue.